spec/features/labels_controller_spec.rb

```diff
@@ -0,0 +1,133 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+describe 'LabelsController', js: true do
+  let(:admin) { create(:admin) }
+  let(:password) { attributes_for(:admin)[:password] }
+
+  before do
+    sign_in_with_js_driver(admin, password)
+  end
+
+  describe '#index' do
+    context 'when using name filter' do
+      let!(:matching_labels) { [create(:label, name: 'Loops'), create(:label, name: 'Java loops')] }
+      let!(:not_matching_labels) { [create(:label, name: 'label1'), create(:label, name: 'python for loop')] }
+
+      it 'filters by name correctly' do
+        visit(labels_path)
+        fill_in id: 'label-name-filter-input', with: 'loops'
+        find('input', id: 'label-name-filter-input').send_keys(:enter)
+        wait_for_ajax
+        matching_labels.each {|label| expect(page).to have_selector('.labels-table > tbody > tr > td > .task_label', text: label.name) }
+        not_matching_labels.each {|label| expect(page).not_to have_selector('.labels-table > tbody > tr > td > .task_label', text: label.name) }
```

Comment:
I though being overspecific was better than being not specific enough to avoid false negatives in testing. I shortened the css a bit to make it more readable.